### ISSUE TITLE
To deploy multiple Model Serving Runtime in NERC RHOAI 

### ISSUE THREAD

**Issue by Milstein** [nerc-project/operations#856]
Can we check and install all these or most of them to our OCP Test and Prod setup:

https://github.com/rh-aiservices-bu/llm-on-openshift/tree/main?tab=readme-ov-file#inference-servers

How to configure is explained [here](https://github.com/eartvit/llm-on-ocp?tab=readme-ov-file#configure-a-serving-runtime).

**Comment by Milstein** [comment]
Our current Model Server only offers "OpenVINO" Serving runtime:

![Image](https://github.com/user-attachments/assets/cbc74b6e-0328-47d0-82ba-81734ead43c4)

**Comment by IsaiahStapleton** [comment]
Also, our test cluster currently has an issue with KServe. When trying to deploy a model the kserve webhook fails to call. 

`ai-performance-profiling               6s          Warning   InternalError                         inferenceservice/granite                                                 fails to reconcile predictor: Internal error occurred: failed calling webhook "webhook.serving.knative.dev": failed to call webhook: Post "https://webhook.knative-serving.svc:443/defaulting?timeout=10s": context deadline 
`

However when inspecting the knative-serving namespace there does not appear to be an issue with the webhook pod so I'm not sure the issue. I was granted access to the Albany cluster and I set up RHOAI from scratch for model serving and have not run into any issues with model serving there, and currently have granite model deployed there.

**Comment by Milstein** [comment]
Also we need to check if this runtime can be added: https://developers.redhat.com/articles/2024/11/22/how-ramalama-makes-working-ai-models-boring#why_not_just_use_ollama_

**Comment by Milstein** [comment]
https://github.com/rh-aiservices-bu/llm-on-openshift/tree/main/serving-runtimes
@IsaiahStapleton Need to add as much as we can.

**Comment by IsaiahStapleton** [comment]
@Milstein I will look into this this week. @hpdempsey those serving runtimes in production I listed above only work for single-modal serving (which is what comes with the default RHOAI configuration), Milson would like to see these serving runtimes available for multi-modal serving.

**Comment by Milstein** [comment]
@IsaiahStapleton @hpdempsey: Can you please confirm if RH doesn't offer any of these extra runtime that support multiple models hosting using same server i.e. multi-model option? 
If that is the case we can go with Single-model based runtime.

**Comment by IsaiahStapleton** [comment]
@Milstein I believe you can just change the value of multimodel in the servingruntime yaml to true in order to support multi-model serving. such as line 28 here: https://github.com/rh-aiservices-bu/llm-on-openshift/blob/main/serving-runtimes/vllm_runtime/vllm-runtime.yaml#L28. I'm not really sure if these serving runtimes are necessarily supported for multi-model serving since they come base configures as single-model serving but i will create a multi-model runtime this way and see if i can successfully deploy a model using it.

**Comment by hpdempsey** [comment]
OK, I misunderstood what the original issue was.   So production RHOAI is set up for single model serving platform, so each model gets its own server.  This should be the best performing option, and  is recommended for production.  Are you saying you want to change production to multi-model serving, and if so why?  It doesn't sound like you can pick both.  
https://docs.redhat.com/en/documentation/red_hat_openshift_ai_self-managed/2-latest/html/serving_models/about-model-serving_about-model-serving#about-model-serving_about-model-serving

**Comment by hpdempsey** [comment]
Following up after more discussion with Isaiah, multi-model serving is more involved than just changin YAML.   Is there a use case we need to implement with multi-model serving @Milstein ?

**Comment by Milstein** [comment]
@hpdempsey @IsaiahStapleton: If the single-model serving platform is recommended for production use for these runtime then let's stick with that. Please let me know once it is ready to be tested on test cluster, I can do some quick test so that we can make them available to the OCP prod as well.

**Comment by morantara** [comment]
@Milstein did you get the information you needed here?

**Comment by morantara** [comment]
@Milstein do you have any updates here?

**Comment by hpdempsey** [comment]
@dystewart @tssala23 please include Milson's request in our plans for the new RHOI on the test cluster that is the basis for our next Production NERC update.

**Comment by morantara** [comment]
@IsaiahStapleton hello! do you think this will be closed within this sprint or should it be pushed to the next one?

**Comment by IsaiahStapleton** [comment]
I have already tested RHOAI 2.19 with OpenShift 2.17 on the Albany cluster.

**Comment by IsaiahStapleton** [comment]
I am closing this issue because I have opened a new issue that when done will address most if not all of the concerns that were brought up in this issue: https://github.com/nerc-project/operations/issues/1104
